packages/data/src/world/mq/ganon_castle_mq.yml

```diff
@@ -35,92 +39,235 @@
+"Ganon Castle Spirit Trial Ending Room":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Spirit Trial Sun Room Sun Side": "true"
+  events:
+    GANON_TRIAL_SPIRIT: "can_use_light_arrows"
+  locations:
+    "MQ Ganon Pot Spirit End 1": "true"
+    "MQ Ganon Pot Spirit End 2": "true"
+"Ganon Castle Shadow Trial Entrance Side":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Middle Beamos Area 1": "event(GANON_MQ_SHADOW_PROGRESSION)"
+    "Ganon Castle Main Room": "true"
+  events:
+    GANON_MQ_SHADOW_BOMBFLOWER: "can_use_bow || (event(GANON_MQ_SHADOW_PROGRESSION) && (can_use_din || has_bombflowers))"
+    GANON_MQ_SHADOW_PROGRESSION: "longshot_anywhere || (has_lens && ((can_hookshot && event(GANON_MQ_SHADOW_BOMBFLOWER)) || has_hover_boots))"
+  locations:
+    "MQ Ganon Castle Shadow Trial Bomb Flower Chest": "event(GANON_MQ_SHADOW_BOMBFLOWER) && (can_hookshot || has_hover_boots)"
+    "MQ Ganon Castle SR Shadow Front-Right": "event(GANON_MQ_SHADOW_PROGRESSION)"
+    "MQ Ganon Castle SR Shadow Front-Center": "event(GANON_MQ_SHADOW_PROGRESSION)"
+    "MQ Ganon Castle Wonder Item": "event(GANON_MQ_SHADOW_BOMBFLOWER) && event(GANON_MQ_SHADOW_PROGRESSION)"
+"Ganon Castle Shadow Trial Middle Beamos Area 1":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Entrance Side": "event(GANON_MQ_SHADOW_PROGRESSION)"
+    "Ganon Castle Shadow Trial Middle Beamos Area 2": "event(GANON_MQ_SHADOW_TORCH_ICE_BLOCK) || has_hover_boots || hookshot_anywhere || (event(GANON_MQ_SHADOW_EYE_CHEST) && can_hookshot)"
+  events:
+    GANON_MQ_SHADOW_TORCH_ICE_BLOCK: "has_fire"
+  locations:
+    "MQ Ganon Castle SR Shadow Middle": "true"
+"Ganon Castle Shadow Trial Middle Beamos Area 2":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Middle Beamos Area 1": "event(GANON_MQ_SHADOW_TORCH_ICE_BLOCK) || has_hover_boots || hookshot_anywhere"
+    "Ganon Castle Shadow Trial Ending Side": "has_lens || hookshot_anywhere"
+  events:
+    GANON_MQ_SHADOW_TORCH_ICE_BLOCK: "has_fire_arrows"
+    GANON_MQ_SHADOW_EYE_CHEST: "can_use_bow || can_use_slingshot"
+  locations:
+    "MQ Ganon Castle Shadow Trial Switch Chest": "event(GANON_MQ_SHADOW_EYE_CHEST)"
+    "MQ Ganon Castle SR Shadow Back-Left": "has_lens || hookshot_anywhere"
+    "MQ Ganon Castle SR Shadow Back-Center": "has_lens"
+"Ganon Castle Shadow Trial Ending Side":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Middle Beamos Area 2": "has_lens || hookshot_anywhere"
+    "Ganon Castle Shadow Trial Ending Room": "silver_rupees_ganon_shadow"
+"Ganon Castle Shadow Trial Ending Room":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Shadow Trial Ending Side": "true"
+  events:
+    GANON_TRIAL_SHADOW: "can_use_light_arrows"
+  locations:
+    "MQ Ganon Pot Shadow End 1": "true"
+    "MQ Ganon Pot Shadow End 2": "true" 
 "Ganon Castle Stairs":
   region: GANON_CASTLE
   exits:
-    "Ganon Castle Main": "true"
+    "Ganon Castle Main Room": "true"
     "Ganon Castle Tower": "true"
```

Comment:
Looks like this was changed in vanilla and not mq. Going to Tower will need to require the barrier event.

Reply:
Added the barrier macro.